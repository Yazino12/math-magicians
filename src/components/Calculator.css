.calculator {
  display: flex;
  flex-direction: column;
  width: 300px;
  min-height: 200px;
  margin: auto;
}

header {
  padding: 0.8em 0.1em;
  background-color: #858693;
  color: white;
}

header .result {
  font-size: 1.2rem;
  text-align: right;
}

.buttons table {
  border-spacing: 0;
  width: 100%;
}

.buttons tbody {
  display: flex;
  flex-direction: column;
}

.buttons tr {
  display: flex;
}

.buttons td {
  font-size: 0.6rem;
  font-weight: 600;
  flex: 1;
  text-align: center;
  padding: 1.5em 1.8em;
  border: 1px solid #d3d3d3;
  background-color: #e0e0e0;
}

.buttons .signs {
  background-color: #f5913e;
}

.buttons .double-sized {
  flex: 2.93;
}
