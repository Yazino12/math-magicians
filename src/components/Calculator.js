import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState({});

  const handleClick = (event) => {
    setState(() => calculate(state, event.target.innerText));
  };

  return (
    <div className="calculator-wrapper">
      <h2>Lets do some Math</h2>
      <div className="calculator">
        <header>
          <p className="result">{state.next || state.total || 0}</p>
        </header>
        <section className="buttons">
          <table>
            <tbody>
              <tr>
                <td onClick={handleClick} role="presentation">
                  AC
                </td>
                <td onClick={handleClick} role="presentation">
                  +/-
                </td>
                <td onClick={handleClick} role="presentation">
                  %
                </td>
                <td className="signs" onClick={handleClick} role="presentation">
                  รท
                </td>
              </tr>
              <tr>
                <td onClick={handleClick} role="presentation">
                  7
                </td>
                <td onClick={handleClick} role="presentation">
                  8
                </td>
                <td onClick={handleClick} role="presentation">
                  9
                </td>
                <td className="signs" onClick={handleClick} role="presentation">
                  x
                </td>
              </tr>
              <tr>
                <td onClick={handleClick} role="presentation">
                  4
                </td>
                <td onClick={handleClick} role="presentation">
                  5
                </td>
                <td onClick={handleClick} role="presentation">
                  6
                </td>
                <td className="signs" onClick={handleClick} role="presentation">
                  -
                </td>
              </tr>
              <tr>
                <td onClick={handleClick} role="presentation">
                  1
                </td>
                <td onClick={handleClick} role="presentation">
                  2
                </td>
                <td onClick={handleClick} role="presentation">
                  3
                </td>
                <td className="signs" onClick={handleClick} role="presentation">
                  +
                </td>
              </tr>
              <tr>
                <td
                  className="double-sized"
                  onClick={handleClick}
                  role="presentation"
                >
                  0
                </td>
                <td onClick={handleClick} role="presentation">
                  .
                </td>
                <td className="signs" onClick={handleClick} role="presentation">
                  =
                </td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  );
}

export default Calculator;
