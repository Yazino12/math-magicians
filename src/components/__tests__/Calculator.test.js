import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../Calculator';

describe('Calculator component test', () => {
  test('Renders', () => {
    const calculator = render(<Calculator />);

    expect(calculator).toMatchSnapshot();
  });

  test('Renders and has last row', () => {
    render(<Calculator />);

    const firstRow = screen.getByRole('row', { name: /0 . =/i });
    expect(firstRow).toBeInTheDocument();
  });
});
