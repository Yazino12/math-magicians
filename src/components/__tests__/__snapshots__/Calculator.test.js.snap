// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component test Renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator-wrapper"
      >
        <h2>
          Lets do some Math
        </h2>
        <div
          class="calculator"
        >
          <header>
            <p
              class="result"
            >
              0
            </p>
          </header>
          <section
            class="buttons"
          >
            <table>
              <tbody>
                <tr>
                  <td
                    role="presentation"
                  >
                    AC
                  </td>
                  <td
                    role="presentation"
                  >
                    +/-
                  </td>
                  <td
                    role="presentation"
                  >
                    %
                  </td>
                  <td
                    class="signs"
                    role="presentation"
                  >
                    รท
                  </td>
                </tr>
                <tr>
                  <td
                    role="presentation"
                  >
                    7
                  </td>
                  <td
                    role="presentation"
                  >
                    8
                  </td>
                  <td
                    role="presentation"
                  >
                    9
                  </td>
                  <td
                    class="signs"
                    role="presentation"
                  >
                    x
                  </td>
                </tr>
                <tr>
                  <td
                    role="presentation"
                  >
                    4
                  </td>
                  <td
                    role="presentation"
                  >
                    5
                  </td>
                  <td
                    role="presentation"
                  >
                    6
                  </td>
                  <td
                    class="signs"
                    role="presentation"
                  >
                    -
                  </td>
                </tr>
                <tr>
                  <td
                    role="presentation"
                  >
                    1
                  </td>
                  <td
                    role="presentation"
                  >
                    2
                  </td>
                  <td
                    role="presentation"
                  >
                    3
                  </td>
                  <td
                    class="signs"
                    role="presentation"
                  >
                    +
                  </td>
                </tr>
                <tr>
                  <td
                    class="double-sized"
                    role="presentation"
                  >
                    0
                  </td>
                  <td
                    role="presentation"
                  >
                    .
                  </td>
                  <td
                    class="signs"
                    role="presentation"
                  >
                    =
                  </td>
                </tr>
              </tbody>
            </table>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-wrapper"
    >
      <h2>
        Lets do some Math
      </h2>
      <div
        class="calculator"
      >
        <header>
          <p
            class="result"
          >
            0
          </p>
        </header>
        <section
          class="buttons"
        >
          <table>
            <tbody>
              <tr>
                <td
                  role="presentation"
                >
                  AC
                </td>
                <td
                  role="presentation"
                >
                  +/-
                </td>
                <td
                  role="presentation"
                >
                  %
                </td>
                <td
                  class="signs"
                  role="presentation"
                >
                  รท
                </td>
              </tr>
              <tr>
                <td
                  role="presentation"
                >
                  7
                </td>
                <td
                  role="presentation"
                >
                  8
                </td>
                <td
                  role="presentation"
                >
                  9
                </td>
                <td
                  class="signs"
                  role="presentation"
                >
                  x
                </td>
              </tr>
              <tr>
                <td
                  role="presentation"
                >
                  4
                </td>
                <td
                  role="presentation"
                >
                  5
                </td>
                <td
                  role="presentation"
                >
                  6
                </td>
                <td
                  class="signs"
                  role="presentation"
                >
                  -
                </td>
              </tr>
              <tr>
                <td
                  role="presentation"
                >
                  1
                </td>
                <td
                  role="presentation"
                >
                  2
                </td>
                <td
                  role="presentation"
                >
                  3
                </td>
                <td
                  class="signs"
                  role="presentation"
                >
                  +
                </td>
              </tr>
              <tr>
                <td
                  class="double-sized"
                  role="presentation"
                >
                  0
                </td>
                <td
                  role="presentation"
                >
                  .
                </td>
                <td
                  class="signs"
                  role="presentation"
                >
                  =
                </td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
